server/swap/swap.go

```diff
@@ -1063,27 +1090,29 @@ func (s *Swapper) checkInactionEventBased() {
 			if match.expiredBy(now) {
 				log.Infof("Revoking match %v at %v because at least one published contract has expired.",
 					match.ID(), match.Status)
-				failMatch(false)
+				deleteMatch(false)
 			}
 		case order.MakerRedeemed:
 			// If the maker has redeemed, the taker can redeem immediately, so
 			// check the timeout against the time the Swapper received the
 			// maker's `redeem` request (and sent the taker's 'redemption').
-			if tooOld(match.makerStatus.redeemSeenTime()) {
-				failMatch(true)
+			if tooOld(match.makerStatus.redeemSeenTime()) { // rlocks swapStatus.mtx
+				deleteMatch(true)
 			}
 		}
 	}
 
+	// Check and delete atomically
 	s.matchMtx.Lock()
-	defer s.matchMtx.Unlock()
-
 	for _, match := range s.matches {
 		checkMatch(match)
 	}
+	s.matchMtx.Unlock()
 
-	for _, match := range deletions {
-		s.deleteMatch(match)
+	// Record failed matches in the DB and auth mgr, unlock coins, and send
+	// revoke_match messsages.
```

Comment:
typo `messsages` (and in another place in this PR),

btw, might be maybe get rid of duplicated bits by something like: https://github.com/chappjc/dcrdex/pull/4

Reply:
Good suggestion!  I'll get this PR out of draft early next week and will include this.